pennylane/gradients/parameter_shift.py

```diff
@@ -174,19 +176,14 @@ def _single_meas_grad(result, coeffs, unshifted_coeff, r0):
 def _multi_meas_grad(res, coeffs, r0, unshifted_coeff, num_measurements):
     """Compute the gradient for multiple measurements by taking the linear combination of
     the coefficients and each measurement result."""
-    g = []
     if r0 is None:
         r0 = [None] * num_measurements
-    for meas_idx in range(num_measurements):
-        # Gather the measurement results
-        meas_result = [param_result[meas_idx] for param_result in res]
-        g_component = _single_meas_grad(meas_result, coeffs, unshifted_coeff, r0[meas_idx])
-        g.append(g_component)
+    if res == ():
+        res = tuple(() for _ in range(num_measurements))
+    return tuple(_single_meas_grad(r, coeffs, unshifted_coeff, r0_) for r, r0_ in zip(res, r0))

```

Comment:
Changing the expected axes from `(parameters, measurements)` to `(measurements, parameters)` allows us to simplify the logic here.

Reply:
Why are we changing the expected axes and where is that happening?

Reply:
**Why:** Before supporting broadcasting, there was a preference for "parameters first".
With broadcasting, "measurements first" is better, so now we have a tie between broadcasting and no broadcasting. However, "measurements first" simplifies a few small things (like the change here), so I went for that one.

Reply:
**Where:** [here](https://github.com/PennyLaneAI/pennylane/pull/5666/files#r1603898487) and [here](https://github.com/PennyLaneAI/pennylane/pull/5666/files#r1603899506)